src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -110,6 +114,15 @@ public int SpectrumCount
         }
 
         /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        public IFileSystem FileSystem { private get; set; }
```

Comment:
Or [this](https://stackoverflow.com/a/26797938) way. Be careful however. Make sure we dont have to destroy provided services manually.

Reply:
Or using [Inject] annotation for property.